lib/travis/build/script/r.rb

```diff
@@ -324,7 +324,7 @@ def r_binary_install(packages)
           return if packages.empty?
           packages = Array(packages)
           if config[:os] == 'linux'
-            unless config[:sudo]
+            unless config[:sudo] and config[:dist] != 'precise'
```

Comment:
Eschew compound conditions with `unless`. It is easier to understand:

```ruby
if !config[:sudo] or config[:dist] == 'precise'
```
than
```ruby
unless config[:sudo] and config[:dist] != 'precise'
```

Also, this new criterion should be mentioned in the message below.